courseraresearchexports/models/ExportRequest.py

```diff
@@ -0,0 +1,371 @@
+            json_request['exportType'] = self.export_type
+        if self.anonymity_level:
+            json_request['anonymityLevel'] = self.anonymity_level
+        if self.statement_of_purpose:
+            json_request['statementOfPurpose'] = self.statement_of_purpose
+        if self.schema_names:
+            json_request['schemaNames'] = self.schema_names
+        if self.interval:
+            json_request['interval'] = self.interval
+        if self.ignore_existing:
+            json_request['ignoreExisting'] = self.ignore_existing
+
+        return json_request
+
+    @classmethod
+    def from_args(cls, **kwargs):
+        """
+        Create a ExportResource object using the parameters required. Performs
+        course_id/partner_id inference if possible.
+        :param kwargs:
+        :return export_request: ExportRequest
+        """
+        if kwargs.get('course_slug') and not kwargs.get('course_id'):
+            kwargs['course_id'] = utils.lookup_course_id_by_slug(
+                kwargs['course_slug'])
+        elif kwargs.get('partner_short_name') and not kwargs.get('partner_id'):
+            kwargs['partner_id'] = utils.lookup_partner_id_by_short_name(
+                kwargs['partner_short_name'])
+
+        return cls(**kwargs)
+
+    @classmethod
+    def from_json(cls, json_request):
+        """
+        Deserialize ExportRequest from json object.
+        :param json_request:
+        :return export_request: ExportRequest
+        """
+        kwargs = {}
+        request_scope = json_request['scope']
+        request_scope_context = request_scope['typeName']
+
+        if request_scope_context == 'courseContext':
+            kwargs['course_id'] = request_scope['definition']['courseId']
+        elif request_scope_context == 'partnerContext':
+            kwargs['partner_id'] = \
+                request_scope['definition']['partnerId']['maestroId']
+        elif request_scope_context == 'groupContext':
+            kwargs['group_id'] = request_scope['definition']['groupId']
+
+        return cls(
+            export_type=json_request.get('exportType'),
+            anonymity_level=json_request.get('anonymityLevel'),
+            statement_of_purpose=json_request.get('statementOfPurpose'),
+            schema_names=json_request.get('schemaNames'),
+            interval=json_request.get('interval'),
+            ignore_existing=json_request.get('ignoreExisting'),
+            **kwargs)
+
+    @property
+    def scope_id(self):
+        """
+        Identifier for the scope, assume that only one of course/partner/group
+        is defined for a valid request.
+        :return scope_id:
+        """
+        return self.course_id or self.partner_id or self.group_id
+
+    @property
+    def schemas(self):
+        """
+        String representation of schemas to be returned in export request.
+        :return schemas:
+        """
+        if self.export_type == EXPORT_TYPE_EVENTING:
+            return 'events'
```

Comment:
It's only used for displaying a list of export requests (`jobs get_all`).  Would it make more sense just to return `None` in those cases?  It looked strange to have that field in the display table blank.